### PR TITLE
feat(install-local.sh): fetch relative package files

### PR DESCRIPTION
**Pull request by oklopfer**
## Purpose

PKGBUILDs can fetch relative files to them relating to the package like so:
Screenshot 2024-03-04 at 2 36 17 PM
Screenshot 2024-03-04 at 2 36 25 PM

We have some packages like [firefox-developer-edition-bin](https://github.com/pacstall/pacstall-programs/tree/master/packages/firefox-developer-edition-bin) that could benefit from this
Screenshot 2024-03-04 at 2 38 17 PM

## Approach

incredibly simple change, if `url` doesn't contain `://`, then set `url` = `$pacstallrepo/packages/$pkgname/$url` and attempt to download it like the rest. 

## Progress

- [x] Do it
- [x] Test it

## Addendum
example script below works
```bash
pkgname="firefox-developer-edition-bin"
maintainer="Oren Klopfer <oren@taumoda.com>"
gives="firefox-developer-edition"
pkgver="122.0b9"
pkgdesc="The browser made for developers. All the latest developer tools in beta, plus experimental features like the Multi-line Console Editor and WebSocket Inspector."
source=(
  "https://download-installer.cdn.mozilla.net/pub/devedition/releases/${pkgver}/linux-x86_64/en-US/firefox-${pkgver}.tar.bz2"
  "${gives}.desktop"
  "${gives}.png::https://www.mozilla.org/media/img/structured-data/logo-firefox-developer.8aedc8028e4d.png"
)
homepage='https://www.mozilla.org/en-US/firefox/developer/'
breaks=("${gives}" "${gives}-git" "${gives}-deb" "${gives}-app")
hash=(
  "8a18a6604183940a67403b21a2e071a510714b502468bdc04624d929655bee5a"
  "SKIP"
  "SKIP"
)
arch=('any')
repology=("project: firefox" "visiblename: firefox-developer-edition")

package() {
  cd ..
  # Make dirs and assign permissions
  sudo mkdir -p "${pkgdir}/opt/firefox-developer-edition"
  sudo cp -r firefox/* "${pkgdir}/opt/firefox-developer-edition"

  # Install the icon for the desktop file
  #wget -q -O "${gives}.png" "${__icon_url}"
  sudo install -Dm644 "firefox/${gives}.png" "${pkgdir}/usr/share/pixmaps/${gives}.png"

  # Install the desktop file
  #wget -q "https://raw.githubusercontent.com/pacstall/pacstall-programs/master/packages/${pkgname}/${gives}.desktop"
  sudo install -Dm644 "firefox/${gives}.desktop" "${pkgdir}/usr/share/applications/${gives}.desktop"
}
```

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -308,6 +308,10 @@ for i in "${!source[@]}"; do
     if [[ -n $PACSTALL_PAYLOAD && ! -f "/tmp/pacstall-pacdeps-$PACKAGE" ]]; then
         dest="${PACSTALL_PAYLOAD##*/}"
     fi
+    if [[ $url != *://* ]]; then
+        # shellcheck disable=SC2031,SC2086
+        url="$(< ${STGDIR}/repo/pacstallrepo)/packages/${pkgname}/${url}"
```
Comment (Elsie19) [inline misc/scripts/install-local.sh:313]: This can have multiple values, each newline separated.

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh (hunk 1 above)
Comment (Elsie19) [reply misc/scripts/install-local.sh]: We discussed in discord about this.

[`4ccf74a` (#276)](https://github.com/pacstall/pacstall/pull/276/commits/4ccf74aaa01b37e5f969d1fb6a39d5c87538040e#diff-3cfe92392f9c3534c9589d9f1f2096e7fbe6d7530ae7eae7312e72bc340ee843R126-R128).

What happens is it's overwriting the array every time, so the fix is to use:

```bash
mapfile -t REPOS -O${#REPOS[@]} <<< "${URLLIST[$IDX]}"
```

Comment (oklopfer) [reply misc/scripts/install-local.sh]: No REPO and REPOS are not the same. This is an unrelated merge.

Comment (oklopfer) [reply misc/scripts/install-local.sh]: See: https://github.com/pacstall/pacstall/pull/1049

**Pull request merged**